### PR TITLE
Use Managed Identity user for migrations

### PR DESCRIPTION
**Pull request by markysoft**
https://eaflood.atlassian.net/browse/PSD-909

### CODE REVIEW

File: src/uk/gov/defra/ffc/Provision.groovy [hunk 1]
```diff
@@ -90,41 +90,67 @@ class Provision implements Serializable {
...
+    ]
+  }
 
-    migrationEnvVars.add("POSTGRES_SCHEMA_ROLE=$schemaRole")
-    migrationEnvVars.add("POSTGRES_SCHEMA_USERNAME=$schemaUser")
-    migrationEnvVars.add("POSTGRES_SCHEMA_NAME=$schemaName")
-    migrationEnvVars.add("POSTGRES_DB=$databaseName")
+  private static def getSchemaToken(ctx, roleName) {
+    def clientId = ctx.sh(returnStdout: true, script: "az identity show --resource-group $ctx.AZURE_POSTGRES_RESOURCE_GROUP --name $roleName --query clientId --output tsv").trim()
+    return ctx.sh(returnStdout: true, script: "curl -s 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fossrdbms-aad.database.windows.net&client_id=$clientId' -H Metadata:true | jq -r .access_token").trim()
```
Comment (johnwatson484) [inline src/uk/gov/defra/ffc/Provision.groovy:125]: Would it be worth adding this URL to an env var?

**Review by johnwatson484 — COMMENTED** [review]
Looks good.  Just a question about whether to store the url for the token as an env var.

Maybe it's not worth it if this is the only place we'll use it and it won't ever change, but urls with dates always worry me.

File: src/uk/gov/defra/ffc/Provision.groovy (hunk 1 above)
Comment (markysoft) [reply src/uk/gov/defra/ffc/Provision.groovy:123]: I think this URL remains the same for all enviroments as it's an internal URL to distribute, i.e. https://serverfault.com/questions/427018/what-is-this-ip-address-169-254-169-254/427022

I think if we switch to using tokens for service bus, that might be a good time to split out the URL.

File: src/uk/gov/defra/ffc/Provision.groovy [hunk 2]
```diff
@@ -37,8 +37,6 @@ class Provision implements Serializable {
 
       ctx.withEnv(getMigrationEnvVars(ctx, environment, repoName, pr)) {
         ctx.dir(migrationFolder) {
-          // migrations may change in different builds of a PR, refresh schema to avoid errors
-          ctx.sh("docker-compose -p $repoName-$pr -f docker-compose.migrate.yaml run schema-down")
```
Comment (markysoft) [inline src/uk/gov/defra/ffc/Provision.groovy:41]: Turns out we were running schema down twice, as it's done earlier in 'deletePrResources. I've removed the duplicate call here.

**Review by johnwatson484 — APPROVED** [review]
LGTM

**Pull request merged**